dimod/constrained.py

```diff
@@ -642,7 +615,8 @@ def set_objective(self, objective: Union[BinaryQuadraticModel, QuadraticModel]):
         """Set the objective of the constrained quadratic model.
```

Comment:
Need to add `Iterable` to `objective: Union[BinaryQuadraticModel, QuadraticModel, Iterable]`